File: archive/p/php/insertion-sort.php
<?php

function insertion_Sort($my_array)
{
	for($i=0;$i<count($my_array);$i++){
		$val = $my_array[$i];
		$j = $i-1;
		while($j>=0 && $my_array[$j] > $val){
			$my_array[$j+1] = $my_array[$j];
			$j--;
		}
		$my_array[$j+1] = $val;
	}
    return $my_array;
}

$test_array = array_map('intval', explode(',', $argv[1]));
$array_size = count($test_array);

if ($array_size <= 1)
{
    exit('Usage: please provide a list of at least two integers to sort in the format "1, 2, 3, 4, 5"');
}

$out = insertion_Sort($test_array);
echo implode(', ', $out);
?>

File: archive/p/perl/insertion-sort.pl
#!/usr/bin/env perl
use v5.42;

use feature qw/keyword_any/;
no warnings 'experimental::keyword_any';

sub usage {
    say 'Usage: please provide a list of at least two integers to sort in the format "1, 2, 3, 4, 5"';
    exit;
}

sub parse_list ($s) {
    return undef unless defined $s;

    my @vals = split /\s*,\s*/, $s;

    return undef if @vals < 2;
    return undef if any { $_ !~ /\A-?\d+\z/ } @vals;

    return [ map 0 + $_, @vals ];
}

sub insertion_sort ($a) {
    for my $i ( 1 .. $#$a ) {
        my $key = $a->[$i];
        my $j   = $i - 1;

        $a->[ $j + 1 ] = $a->[$j], $j-- while $j >= 0 && $a->[$j] > $key;
        $a->[ $j + 1 ] = $key;
    }

    return $a;
}

my ($input) = @ARGV;
my $a = parse_list($input) or usage();

insertion_sort($a);
say join ', ', @$a;
